packages/camera/src/components/Camera/index.js

```diff
@@ -1,73 +1,220 @@
+        const [testB, streamB] = await resultB;
+
+        if (streamA.error) { return resultB; }
+        if (streamB.error) { return resultA; }
+
+        return testA.width > testB.width ? resultA : resultB;
+      });
+
+    return {
+      test,
+      stream,
+      constraint,
+      ask: `${test.width}x${test.height}`,
+      browserVer: `${adapter.browserDetails.browser} ${adapter.browserDetails.version}`,
+    };
+  }, [gum]);
+
+  const findDevices = useCallback(async () => {
+    const constraints = { audio: false, video: { facingMode: 'environment' } };
+    const mediaDevices = await navigator.mediaDevices.enumerateDevices();
+    window.stream = await getUserMedia(constraints);
+
+    return mediaDevices.filter(({ kind }) => kind === 'videoinput');
   }, []);
 
-  if (permissions.isGranted === null) {
-    return <View />;
-  }
+  useLayoutEffect(() => {
+    (async () => {
+      if (isEmpty(camera) && !loading) {
+        const video = videoEl.current;
+        setLoading(true);
+
+        const canvas = canvasEl.current;
+        const devices = await findDevices();
+        const bestCandidate = await findBestCandidate(devices);
+
+        setRatio(bestCandidate.test.ratio);
+
+        if ('srcObject' in video) {
+          video.srcObject = bestCandidate.stream;
+        } else {
+          video.src = window.URL.createObjectURL(bestCandidate.stream);
+        }
+
+        video.onloadedmetadata = () => { video.play(); };
+
+        canvas.width = bestCandidate.test.width;
+        canvas.height = bestCandidate.test.height;
+
+        const takePictureAsync = () => {
+          canvas.getContext('2d').drawImage(video, 0, 0, canvas.width, canvas.height);
+          const imageType = utils.getOS() === 'ios' ? 'image/png' : 'image/webp';
+          return { uri: canvasEl.current.toDataURL(imageType) };
+        };
+
+        const cameraRef = { takePictureAsync, canvas, video };
+        if (typeof onRef === 'function') { onRef(cameraRef); }
+
+        setCamera(cameraRef);
+        if (typeof onCameraReady === 'function') { onCameraReady(); }
 
-  if (permissions.isGranted === false || !available) {
-    return <Text>No access to camera</Text>;
-  }
+        setLoading(false);
+      }
+    })();
+  }, [camera, findBestCandidate, findDevices, loading, onCameraReady, onRef]);
 
   return (
     <View
       accessibilityLabel="Camera container"
-      style={[containerStyle, size]}
+      style={[styles.container, containerStyle, { width: videoWith, height: videoHeight }]}
     >
-      <ExpoCamera
-        ref={onRef}
-        ratio={ratio}
-        onMountError={handleError}
-        {...passThroughProps}
-      >
-        {children}
-      </ExpoCamera>
-      {title !== '' && <Text style={styles.title}>{title}</Text>}
+      {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
+      <video
+        autoPlay={utils.getOS() === 'ios'}
+        width={videoWith}
+        height={videoHeight}
+        ref={videoEl}
```

Comment:
Have you tried adding `playsInline` (set to true for ios)?

Reply:
Yes, it doesn't work either